### PR TITLE
Massive refactor

### PR DESCRIPTION
**Pull request by mgabrel**
I still plan to work on this branch some more, but most of the work is done so putting this up for review

### CODE REVIEW

**Comment by mgabrel** [comment]
This branch does not produce output correctly for FileType 3

File: src/generate_output.py [hunk 1]
```diff
@@ -92,195 +65,106 @@ def wrap_commas(value):
...
+def adjust_batch_date_based_on_undocumented_rules(batch_date):
+  return (pd(int(batch_date.split('-')[0]), int(batch_date.split('-')[1]), int(batch_date.split('-')[2])) + 3 * business_day_ounce).strftime('%Y-%m-%d')
+
+def generate_output_rows(consolidated_rows_dict, output_rows_dict):
+  for filetype in consolidated_rows_dict:
+    for year_type in consolidated_rows_dict[filetype]:
+      for fiscal_year in consolidated_rows_dict[filetype][year_type]:
+        running_total = Decimal(0.0000)
+        batch_id = ''
+        batch_date = ''
```
Comment (mgabrel) [inline src/generate_output.py]: These two values can be passed in from the main method

File: src/generate_output.py [hunk 2]
```diff
@@ -92,195 +65,106 @@ def wrap_commas(value):
...
\ No newline at end of file
+    if year_type == NEXT_YEAR:
+      return []
+    else:
+      return [[posting_date(batch_date, fiscal_year), 'Batch ID: ' + batch_id, '1820', UNRESTRICTED, '', '', '', '', '', '', '', '', '', '', balancing_row_description(batch_date, batch_id), print_running_total(running_total, True), '', '', '\n']]
+
+def adjust_batch_date_based_on_undocumented_rules(batch_date):
+  return (pd(int(batch_date.split('-')[0]), int(batch_date.split('-')[1]), int(batch_date.split('-')[2])) + 3 * business_day_ounce).strftime('%Y-%m-%d')
+
+def generate_output_rows(consolidated_rows_dict, output_rows_dict):
```
Comment (mgabrel) [inline src/generate_output.py]: This method needs to be split up in order to account for differing logic for Filetype1/2 VS FileType3

File: src/calculate_totals.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    for year_case in consolidated_rows[filetype]:
+      for fiscal_year in consolidated_rows[filetype][year_case]:
+        for consolidation_key in consolidated_rows[filetype][year_case][fiscal_year]:
+          sae_row      = consolidated_rows[filetype][year_case][fiscal_year][consolidation_key][SAE_ROW]
+          if sae_row[PAYMENT_CODE] == '':
+            sae_row[PAYMENT_CODE] = '***Blank***'
+          expense_type = consolidated_rows[filetype][year_case][fiscal_year][consolidation_key][EXPENSE_TYPE]
+          sum          = consolidated_rows[filetype][year_case][fiscal_year][consolidation_key][SUM]
+          totals[filetype]['Grand'] += sum
+          if filetype == 'FileType1' or filetype == 'FileType2':
```
Comment (mgabrel) [inline src/calculate_totals.py]: Refactor 'FileTypeX' strings into global_var

**Comment by avinewman** [comment]
please delete this p-req and repost it when it's producing proper output

**Pull request closed without merging**